packages/nimble-components/src/patterns/button/styles.ts

```diff
@@ -78,26 +81,56 @@ export const styles = css`
     }
 
     .control:hover {
-        box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-        outline: none;
+        border-color: ${borderHoverColor};
+        box-shadow: 0px 0px 0px 1px ${borderHoverColor} inset;
+        transition: box-shadow ${mediumDelay};
     }
 
     .control${focusVisible} {
+        border-color: ${borderHoverColor};
         box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-        outline: ${borderWidth} solid ${borderHoverColor};
-        outline-offset: -4px;
     }
 
     .control:active {
         box-shadow: none;
         outline: none;
+        transition: outline ${mediumDelay};
+        transition: box-shadow ${mediumDelay};
+    }
+
+    .control::before {
+        content: '';
+        position: absolute;
+        width: 100%;
+        height: 100%;
+        pointer-events: none;
+        box-sizing: border-box;
+        outline: none;
+        background-clip: content-box;
+        z-index: -1;
+    }
+
+    .control::before${focusVisible} {
+        outline: ${borderWidth} solid ${borderHoverColor};
+        outline-offset: -3px;
+        transition: outline 0.5s;
+        color: transparent;
+    }
+
+    .control::before:active {
+        outline: none;
     }
 
     :host([disabled]) .control {
         box-shadow: none;
         outline: none;
     }
```

Comment:
The selection target of this element is `.control` and should be grouped with the other `.control` targetted selectors.

Reply:
Got it. Also, it seems like declaring css classes with `::before` in the most recent commit's format (ex. `.control::before${focusVisible}`) causes the css classes to not work. Looking at other places that use `::before` like the  [nimble tab styles.ts](https://github.com/ni/nimble/blob/631614c7c6fb333e4018058611c9986d61b0002c/packages/nimble-components/src/patterns/tab/styles.ts), it seems that they use the format `.control${focusVisible}::before`, so I'll switch back to that format since when I do it fixes the problem of the css class not working.

Reply:
Ah yea makes sense. A real element has focus states, the psedoelements `::before` and `::after` don't so yea you want to target the `::before` pseudelement of the `.control` element when the `.control` element has focus. `.control${focusVisible}::before`